cobc/parser.y

```diff
@@ -314,24 +314,20 @@ static COB_INLINE void
 check_area_a (cb_tree stmt) {
 	if (!cobc_in_area_a && cobc_areacheck) {
```

Comment:
Should we still use `cobc_areacheck` here? I think the correct way is to check the format, no?

Reply:
`cobc_areacheck` itself is set based on the format: in `cobc_set_source_format` (`pplex.l`), or else in the scanner when `#(NO)AREACHECK` is seen, filtered using `cobc_has_areacheck_directive` (`pplex.l` too).

Reply:
Hm, so long as  `cobc -fformat=cobol85 prog.cob` raises the messages (=-fformat also sets the areacheck value) that's fine; this would allow both `cobc -std=cobol85` and `cobc -fformat=cobol85` raising these expected messages and both of these with `-fno-areacheck` to not raise them.

But then `cobc -fareacheck` would do nothing... also fine as long as this is clear from the docs.

Reply:
May renaming the `areacheck` option into `default-areacheck` or `areacheck-by-default` be clearer?

Reply:
I mean, as the effect of this option is both: dependent on the format, and can be overridden by `$SET` directives.

Reply:
> can be overridden by `$SET` directives.

Last time I've checked they were only active if the format has the area a; which is reasonable.

> May renaming the `areacheck` option [...] would be clearer?

possibly yes, not sure about a reasonable name, though (note: we do not necessarily name that "areacheck" at all.

Reply:
Using `areacheck` in the name of the option has the advantage that it easily relates to the `AREACHECK` directive.

Reply:
In any case, I've extended its documentation to state what this entails. Maybe too much for a compiler flag though.

Reply:
I think that's OK (I've pushed a minor edit because of gnucobol.pdf). I'm ok with using that name (there are some days left, in case we come up with a better name) ;-)